### ISSUE TITLE
How to render a normal map with vispy?

### ISSUE THREAD

**Issue by wangg12** [vispy/vispy#1615]
I found a tutorial to render normal map with python opengl
[https://github.com/jcteng/python-opengl-tutorial/blob/master/tu_10_normal_mapping.py](https://github.com/jcteng/python-opengl-tutorial/blob/master/tu_10_normal_mapping.py),
however the raw opengl APIs looks very complex and hard to understand. Is there a simple example of how to render a normal map with vispy?

**Comment by djhoese** [comment]
I don't think we have any examples right now that specifically show normals from a given object. We do have utilities for loading different object formats (I think). It is really hard to tell everything that that code is doing due to the lack of documentation.

**Comment by wangg12** [comment]
It would be great if there is an example for rendering the normal map with a .ply file.

**Comment by djhoese** [comment]
This is not something I am familiar with. Since this seems like something you are interested in it would be great if you could add the example and make a pull request to share with others.

**Comment by kmuehlbauer** [comment]
@wangg12 After having a quick look at the provided example I support @djhoese. An example would be really nice. vispy should have all the needed machinery (`Texture2D` to load the normal map). Everything else could be done in the shaders first. Going further we could think about how and where to integrate this. But a working example would be the initial step.

**Comment by wangg12** [comment]
I am also not familiar with this. My problem is that, given an .ply object which contains the surface(vertex) normals, I want to render out a surface normal map with a given pose. I still don’t know how this could be achieved with vispy. Any hints would be helpful.

**Comment by GuillaumeFavelier** [comment]
I made a really rough [gist example](https://gist.github.com/GuillaumeFavelier/cd22bbbf0116c9237be6be1bf160618d) demonstrating [normal mapping](https://en.wikipedia.org/wiki/Normal_mapping), there is definitely room for improvement and I agree with @kmuehlbauer, the `Texture2D` idea sounds nice.

**Comment by GuillaumeFavelier** [comment]
This is horrible but this what I get xD

![image](https://user-images.githubusercontent.com/18143289/56047480-c3c36580-5d45-11e9-8a9a-6a1abedf5034.png)

**Comment by GuillaumeFavelier** [comment]
Here is [another example](https://gist.github.com/GuillaumeFavelier/d12b3f9406b68b11bcae09bddfdbaf3e) this time using `Texture2D` but still very straightforward with a plane using normals towards the positive z direction.

![image](https://user-images.githubusercontent.com/18143289/56137489-43933f00-5f95-11e9-8d47-bf7066037cf2.png)
